libflux/src/flux/semantic/nodes.rs

```diff
@@ -976,36 +976,50 @@ impl BinaryExpr {
                 Constraint::Kind(self.typ.clone(), Kind::Divisible),
             ]),
             ast::Operator::GreaterThanOperator => Constraints::from(vec![
-                Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
+                // https://github.com/influxdata/flux/issues/2393
+                // Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
                 Constraint::Equal(self.typ.clone(), MonoType::Bool),
                 Constraint::Kind(self.left.type_of().clone(), Kind::Comparable),
+                Constraint::Kind(self.right.type_of().clone(), Kind::Comparable),
```

Comment:
This change probably should have been pointed directly at master since it's not `use-algow` specific